### ISSUE TITLE
Crash from adding List<int> to Dictionary (regression starting from 3.2.2)

### ISSUE THREAD

**Issue by ondesic** [godotengine/godot#40023]
I finally was able to get 3.2.2 configured to work but my program would crash without any log on startup.  I narrowed it down to one line of code.
In 3.2.1 this code ran fine, however, with no code change and moving to 3.2.2, the app crashes.  Here is the test code:

```
public List<string> TheName = new List<string>();
public List<int> ClassLevel = new List<int>();

    public override void _Ready()
    {
        TheName.Add("Sampson");
        ClassLevel.Add(12);

        Godot.Collections.Dictionary d = new Godot.Collections.Dictionary();

        d.Add("name", TheName);
        d.Add("sd", ClassLevel);
        GD.Print("Hello");
    }
```

The program exits without warnings or error at the line `d.Add("sd", ClassLevel);`
If the ClassLevel list is empty it will not crash.  However, once I add any integer to the list, it cannot be added to the dictionary.
No other kind of list seems to crash.  I included a `List<string> `to show that Godot can add other types of lists just fine.

**Comment by FlashDaggerX** [comment]
@ondesic If you run the editor in a shell, mono should print an error dump into stdout. I got an error when running it too, but I just wanna see if it's the same as yours.

```
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x7fc67a14d625 - /usr/lib/libmonosgen-2.0.so.1 : (null)
	0x7fc67a14d9bc - /usr/lib/libmonosgen-2.0.so.1 : (null)
	0x7fc67a0d8b82 - /usr/lib/libmonosgen-2.0.so.1 : (null)
	0x7fc67a147161 - /usr/lib/libmonosgen-2.0.so.1 : (null)
	0x7fc67a253674 - /usr/lib/libmonosgen-2.0.so.1 : mono_object_get_class
	0x7ffd82b49240 - Unknown

=================================================================
	Telemetry Dumper:
=================================================================
Pkilling 0x7fc62c40b700 from 0x7fc678c74840
Pkilling 0x7fc62c20a700 from 0x7fc678c74840
Entering thread summarizer pause from 0x7fc678c74840
Finished thread summarizer pause from 0x7fc678c74840.
Could not exec mono-hang-watchdog, expected on path '/etc/../bin/mono-hang-watchdog' (errno 2)
debugger-agent: Unable to listen on 9

Waiting for dumping threads to resume

=================================================================
	External Debugger Dump:
=================================================================
mono_gdb_render_native_backtraces not supported on this platform, unable to find gdb or lldb

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x7fc67a253674):0x7fc67a253664  89 df 48 89 1c 24 48 89 44 24 08 e8 cc 6d 0e 00  ..H..$H.D$...m..
0x7fc67a253674  48 8b 55 00 48 89 de 48 89 c7 48 8b 2a e8 4a 6f  H.U.H..H..H.*.Jo
0x7fc67a253684  0e 00 48 8b 4c 24 18 64 48 33 0c 25 28 00 00 00  ..H.L$.dH3.%(...
0x7fc67a253694  75 0a 48 83 c4 28 48 89 e8 5b 5d c3 e8 3b 56 de  u.H..(H..[]..;V.

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at Godot.Collections.Dictionary:godot_icall_Dictionary_Add <0x000d0>
	  at Godot.Collections.Dictionary:Add <0x0009b>
	  at Node:_Ready <0x001ff>
	  at System.Object:runtime_invoke_void__this__ <0x00170>
=================================================================
```

**Comment by FlashDaggerX** [comment]
A bisect shows this as the first bad commit (3.2 branch, commit 6aca9326):

```
commit 6aca932659d46c293328f6e408667123cac03d02 (HEAD, refs/bisect/bad)
Author: Ignacio Etcheverry <ignalfonsore@gmail.com>
Date:   Thu Apr 23 02:19:32 2020 +0200

    C#: Revert marshalling of IDictionary/IEnumerable implementing types
    
    Added marshalling for `System.Collections.Generic.List<T>` and
    `System.Collections.Generic.Dictionary<TKey, TValue>`.
```

**Comment by timothyparez** [comment]
Is there any chance of this getting fixed in 3.2.x?
This breaking change basically cripples decent C# support.

**Comment by timothyparez** [comment]
This was fixed in 3.2.4-rc2 and is back in 3.2.4-rc3
I am trying to see if I can get more info
